### PR TITLE
Landmine stepoff

### LINKED ISSUE #21658: Landmines should only explode after you step off them
Title.

### PR DESCRIPTION
**Pull request by YuriyKiss**
## About the PR
Landmine will explode when player steps off it. Landmine will make a sound effect when player steps on it
Close #21658 (Issue has some other suggestions, turn them into separate issues if relevant)

Requires https://github.com/space-wizards/RobustToolbox/pull/4883

## Why / Balance
IRL some landmines will only trigger after you release the pressure put on them.

## Technical details
There are two landmine modes now, one will make landmine trigger immediately after player steps on them another one will only trigger after player steps off the landmine (this is the default now).

## Media
Landmine stepoff in action:

https://github.com/space-wizards/space-station-14/assets/59531932/6e884619-7217-4301-bd78-6e843e0d78f1

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**
- tweak: landmines will trigger after you step off them
- add: landmine trigger sound

### CODE REVIEW

**Comment by metalgearsloth** [comment]
Should play a beep when you step on it.

**Comment by YuriyKiss** [comment]
Sure, what I can't handle myself is - which sound should I use for "beep"? If there's nothing ready - I can look for it myself but I don't know how to integrate that sound into game. In that case tell me where I can look up the implementation.

**Comment by YuriyKiss** [comment]
Final result:
https://github.com/space-wizards/space-station-14/assets/59531932/e9c10f74-2a30-4269-9dba-ab11345728f9

File: Content.Server/LandMines/LandMineComponent.cs [hunk 1]
```diff
@@ -1,6 +1,23 @@
...
+    /// True - StepTrigger, False - StepOffTrigger
+    /// </summary>
+    [DataField, ViewVariables(VVAccess.ReadWrite)]
+    public bool ExplodeImmediately = false;
+
+    /// <summary>
+    /// Distance at which landmine beep sound can be heard by players
+    /// </summary>
+    [ViewVariables(VVAccess.ReadWrite)]
+    public readonly float TriggerAudioRange = 10f;
```
Comment (deltanedas) [inline Content.Server/LandMines/LandMineComponent.cs:17]: no readonly + datafield

File: Content.Server/LandMines/LandMineComponent.cs [hunk 2]
```diff
@@ -1,6 +1,23 @@
...
+
+    /// <summary>
+    /// Landmine beep SFX path in Resources folder
+    /// </summary>
+    public readonly string MineBeepAudioPath = "/Audio/Effects/beep_landmine.ogg";
```
Comment (deltanedas) [inline Content.Server/LandMines/LandMineComponent.cs:22]: change to SoundSpecifier remove readonly and add datafield

File: Content.Server/LandMines/LandMineSystem.cs [hunk 3]
```diff
@@ -1,43 +1,75 @@
...
-                args.Tripper,
+                tripper,
                 PopupType.LargeCaution);
-        }
     }
-}
 
+    private void PlayLandmineActivatedSound(EntityUid uid, LandMineComponent component)
+    {
+        var xform = _entityManager.GetComponent<TransformComponent>(uid);
```
Comment (deltanedas) [inline Content.Server/LandMines/LandMineSystem.cs:62]: use proxy Transform + all this should be useless if the sound still gets played after mine is deleted, 1 line to play sound on entity uid. if not then play at coords

File: Content.Shared/StepTrigger/Components/StepTriggerComponent.cs [hunk 4]
```diff
@@ -21,6 +21,12 @@ public sealed partial class StepTriggerComponent : Component
     [ViewVariables, AutoNetworkedField]
     public HashSet<EntityUid> CurrentlySteppedOn = new();
 
+    /// <summary>
+    ///     List of entities that left collision this frame.
+    /// </summary>
+    [ViewVariables, AutoNetworkedField]
```
Comment (deltanedas) [inline Content.Shared/StepTrigger/Components/StepTriggerComponent.cs:27]: datafield so at least this will save

**Review by deltanedas — CHANGES_REQUESTED** [review]

**Review by metalgearsloth — CHANGES_REQUESTED** [review]
I forgor to submit.

File: Content.Server/LandMines/LandMineComponent.cs (hunk 1 above)
Comment (metalgearsloth) [reply Content.Server/LandMines/LandMineComponent.cs:17]: Just use MaxDistance on the soundspecifier audioparams.

File: Content.Server/LandMines/LandMineSystem.cs (hunk 3 above)
Comment (YuriyKiss) [reply Content.Server/LandMines/LandMineSystem.cs]: For now, I saw there's a function to PlayPvs() and used that; will that work as intended with maxDistance set in audioparams?

Also, it's pretty neat that sound ends as soon as landmine is destroyed, that's a logical thing to happen 👍🏼

File: Content.Shared/StepTrigger/Components/StepTriggerComponent.cs (hunk 4 above)
Comment (YuriyKiss) [reply Content.Shared/StepTrigger/Components/StepTriggerComponent.cs]: I am not sure about this one; two other HashSets with similar usage does not use DataField, but I guess fine, it's easy to add and remove such things

File: Content.Server/LandMines/LandMineComponent.cs [hunk 5]
```diff
@@ -1,6 +1,20 @@
...
+    /// </summary>
+    [DataField, ViewVariables(VVAccess.ReadWrite)]
+    public bool TriggerImmediately = false;
+
+    /// <summary>
+    /// Trigger sound effect when stepping onto landmine
+    /// </summary>
+    [DataField("sound")]
+    public SoundSpecifier TriggerSound;
 }
```
Comment (metalgearsloth) [inline Content.Server/LandMines/LandMineComponent.cs:20]: If a field is not required then it should be nullable.

File: Content.Server/LandMines/LandMineSystem.cs [hunk 6]
```diff
@@ -1,43 +1,69 @@
...
+using Robust.Server.GameObjects;
+using Robust.Shared.Audio.Systems;
+using Robust.Shared.Map;
 using Robust.Shared.Player;
 
 namespace Content.Server.LandMines;
 
 public sealed class LandMineSystem : EntitySystem
 {
+    [Dependency] private readonly IEntityManager _entityManager = default!;
```
Comment (metalgearsloth) [inline Content.Server/LandMines/LandMineSystem.cs:13]: This is already a dependency on systems.

File: Resources/Audio/Effects/attributions.yml [hunk 7]
```diff
@@ -201,3 +201,8 @@
   copyright: Bhijn and Myr (github username deathride58)
   license: CC0-1.0
   source: https://github.com/space-wizards/space-station-14/pull/23212
+
+- files: [beep_landmine.ogg]
+  copyright: '"beep_landmine.ogg" by kaktuscsc of Discord for SS14'
+  license: "CC-BY-SA-3.0"
+  source: https://cdn.discordapp.com/attachments/1192464327228260382/1192464555079647365/beep_landmine.ogg?ex=65a92c28&is=6596b728&hm=19c9d95e754568bfb01be57e4240ef281de6751b2fdc47659d74470a79ed1503&
```
Comment (metalgearsloth) [inline Resources/Audio/Effects/attributions.yml:208]: Don't link to discord attachments as links can die at any time.

File: Content.Shared/StepTrigger/Systems/StepTriggerSystem.cs [hunk 8]
```diff
@@ -78,6 +80,18 @@ private bool Update(EntityUid uid, StepTriggerComponent component, TransformComp
...
+        // We shouldn't raise this event in OnEndCollide - game shits itself once in a while
+        foreach (var otherUid in component.StoppedColliding)
+        {
+            var evStepOff = new StepOffTriggeredEvent { Source = uid, Tripper = otherUid };
+            RaiseLocalEvent(uid, ref evStepOff);
+        }
+
+        component.StoppedColliding.Clear();
+    }
+
```
Comment (metalgearsloth) [inline Content.Shared/StepTrigger/Systems/StepTriggerSystem.cs:94]: Use EndCollide.

File: Content.Shared/StepTrigger/Components/StepTriggerComponent.cs [hunk 9]
```diff
@@ -21,6 +21,12 @@ public sealed partial class StepTriggerComponent : Component
     [ViewVariables, AutoNetworkedField]
     public HashSet<EntityUid> CurrentlySteppedOn = new();
 
+    /// <summary>
+    ///     List of entities that left collision this frame.
+    /// </summary>
+    [DataField, ViewVariables, AutoNetworkedField]
+    public HashSet<EntityUid> StoppedColliding = new();
+
```
Comment (metalgearsloth) [inline Content.Shared/StepTrigger/Components/StepTriggerComponent.cs:29]: This doesn't need to exist for a server-only system.

**Review by metalgearsloth — CHANGES_REQUESTED** [review]
Just have StepTrigger have a bool on whether trigger is on startcollide or endcollide instead as it will be much cleaner.

**Comment by YuriyKiss** [comment]
https://github.com/space-wizards/RobustToolbox/pull/4883 was updated recently, after it is merged this PR is finished

**Comment by Arteben** [comment]
When a regular mine is triggered, after the title, when you step off this mine, an error occurs. This is not a mod release.

<details>
  <summary>The error</summary>
  [FATL] unhandled: Robust.Shared.Utility.DebugAssertException: Exception of type 'Robust.Shared.Utility.DebugAssertException' was thrown.
   at Robust.Shared.Utility.DebugTools.Assert(Boolean condition) in /media/artem/Work/dot-net/space-station-wizards-14/RobustToolbox/Robust.Shared/Utility/DebugTools.cs:line 37
   at Robust.Shared.Physics.Systems.SharedPhysicsSystem.DestroyContacts(PhysicsComponent body) in /media/artem/Work/dot-net/space-station-wizards-14/RobustToolbox/Robust.Shared/Physics/Systems/SharedPhysicsSystem.Components.cs:line 199
   at Robust.Shared.Physics.Systems.SharedPhysicsSystem.OnContactEntityQueueDel(EntityUid obj) in /media/artem/Work/dot-net/space-station-wizards-14/RobustToolbox/Robust.Shared/Physics/Systems/SharedPhysicsSystem.Contacts.cs:line 195
   at Robust.Shared.GameObjects.EntityManager.QueueDeleteEntity(Nullable`1 uid) in /media/artem/Work/dot-net/space-station-wizards-14/RobustToolbox/Robust.Shared/GameObjects/EntityManager.cs:line 627
   at Content.Server.Explosion.EntitySystems.ExplosionSystem.TriggerExplosive(EntityUid uid, ExplosiveComponent explosive, Boolean delete, Nullable`1 totalIntensity, Nullable`1 radius, Nullable`1 user) in /media/artem/Work/dot-net/space-station-wizards-14/Content.Server/Explosion/EntitySystems/ExplosionSystem.cs:line 181
   at Content.Server.Explosion.EntitySystems.TriggerSystem.HandleExplodeTrigger(EntityUid uid, ExplodeOnTriggerComponent component, TriggerEvent args) in /media/artem/Work/dot-net/space-station-wizards-14/Content.Server/Explosion/EntitySystems/TriggerSystem.cs:line 134
   at Robust.Shared.GameObjects.EntityEventBus.<>c__DisplayClass48_0`2.<SubscribeLocalEvent>g__EventHandler|0(EntityUid uid, IComponent comp, TEvent& args) in /media/artem/Work/dot-net/space-station-wizards-14/RobustToolbox/Robust.Shared/GameObjects/EntityEventBus.Directed.cs:line 249
   at Robust.Shared.GameObjects.EntityEventBus.<>c__DisplayClass62_0`1.<EntSubscribe>b__0(EntityUid uid, IComponent comp, Unit& ev) in /media/artem/Work/dot-net/space-station-wizards-14/RobustToolbox/Robust.Shared/GameObjects/EntityEventBus.Directed.cs:line 432
   at Robust.Shared.GameObjects.EntityEventBus.EntDispatch(EntityUid euid, Type eventType, Unit& args) in /media/artem/Work/dot-net/space-station-wizards-14/RobustToolbox/Robust.Shared/GameObjects/EntityEventBus.Directed.cs:line 584
   at Robust.Shared.GameObjects.EntityEventBus.RaiseLocalEventCore(EntityUid uid, Unit& unitRef, Type type, Boolean broadcast) in /media/artem/Work/dot-net/space-station-wizards-14/RobustToolbox/Robust.Shared/GameObjects/EntityEventBus.Directed.cs:line 217
   at Robust.Shared.GameObjects.EntityEventBus.RaiseLocalEvent[TEvent](EntityUid uid, TEvent args, Boolean broadcast) in /media/artem/Work/dot-net/space-station-wizards-14/RobustToolbox/Robust.Shared/GameObjects/EntityEventBus.Directed.cs:line 177
   at Content.Server.Explosion.EntitySystems.TriggerSystem.Trigger(EntityUid trigger, Nullable`1 user) in /media/artem/Work/dot-net/space-station-wizards-14/Content.Server/Explosion/EntitySystems/TriggerSystem.cs:line 234
   at Content.Server.LandMines.LandMineSystem.HandleStepTriggered(EntityUid uid, LandMineComponent component, StepTriggeredEvent& args) in /media/artem/Work/dot-net/space-station-wizards-14/Content.Server/LandMines/LandMineSystem.cs:line 27
   at Robust.Shared.GameObjects.EntityEventBus.<>c__DisplayClass50_0`2.<SubscribeLocalEvent>g__EventHandler|0(EntityUid uid, IComponent comp, TEvent& args) in /media/artem/Work/dot-net/space-station-wizards-14/RobustToolbox/Robust.Shared/GameObjects/EntityEventBus.Directed.cs:line 282
   at Robust.Shared.GameObjects.EntityEventBus.<>c__DisplayClass62_0`1.<EntSubscribe>b__0(EntityUid uid, IComponent comp, Unit& ev) in /media/artem/Work/dot-net/space-station-wizards-14/RobustToolbox/Robust.Shared/GameObjects/EntityEventBus.Directed.cs:line 432
   at Robust.Shared.GameObjects.EntityEventBus.EntDispatch(EntityUid euid, Type eventType, Unit& args) in /media/artem/Work/dot-net/space-station-wizards-14/RobustToolbox/Robust.Shared/GameObjects/EntityEventBus.Directed.cs:line 584
   at Robust.Shared.GameObjects.EntityEventBus.RaiseLocalEventCore(EntityUid uid, Unit& unitRef, Type type, Boolean broadcast) in /media/artem/Work/dot-net/space-station-wizards-14/RobustToolbox/Robust.Shared/GameObjects/EntityEventBus.Directed.cs:line 217
   at Robust.Shared.GameObjects.EntityEventBus.RaiseLocalEvent[TEvent](EntityUid uid, TEvent& args, Boolean broadcast) in /media/artem/Work/dot-net/space-station-wizards-14/RobustToolbox/Robust.Shared/GameObjects/EntityEventBus.Directed.cs:line 195
   at Robust.Shared.GameObjects.EntitySystem.RaiseLocalEvent[TEvent](EntityUid uid, TEvent& args, Boolean broadcast) in /media/artem/Work/dot-net/space-station-wizards-14/RobustToolbox/Robust.Shared/GameObjects/EntitySystem.cs:line 160
   at Content.Shared.StepTrigger.Systems.StepTriggerSystem.OnEndCollide(EntityUid uid, StepTriggerComponent component, EndCollideEvent& args) in /media/artem/Work/dot-net/space-station-wizards-14/Content.Shared/StepTrigger/Systems/StepTriggerSystem.cs:line 171
   at Robust.Shared.GameObjects.EntityEventBus.<>c__DisplayClass50_0`2.<SubscribeLocalEvent>g__EventHandler|0(EntityUid uid, IComponent comp, TEvent& args) in /media/artem/Work/dot-net/space-station-wizards-14/RobustToolbox/Robust.Shared/GameObjects/EntityEventBus.Directed.cs:line 282
   at Robust.Shared.GameObjects.EntityEventBus.<>c__DisplayClass62_0`1.<EntSubscribe>b__0(EntityUid uid, IComponent comp, Unit& ev) in /media/artem/Work/dot-net/space-station-wizards-14/RobustToolbox/Robust.Shared/GameObjects/EntityEventBus.Directed.cs:line 432
   at Robust.Shared.GameObjects.EntityEventBus.EntDispatch(EntityUid euid, Type eventType, Unit& args) in /media/artem/Work/dot-net/space-station-wizards-14/RobustToolbox/Robust.Shared/GameObjects/EntityEventBus.Directed.cs:line 584
   at Robust.Shared.GameObjects.EntityEventBus.RaiseLocalEventCore(EntityUid uid, Unit& unitRef, Type type, Boolean broadcast) in /media/artem/Work/dot-net/space-station-wizards-14/RobustToolbox/Robust.Shared/GameObjects/EntityEventBus.Directed.cs:line 217
   at Robust.Shared.GameObjects.EntityEventBus.RaiseLocalEvent[TEvent](EntityUid uid, TEvent& args, Boolean broadcast) in /media/artem/Work/dot-net/space-station-wizards-14/RobustToolbox/Robust.Shared/GameObjects/EntityEventBus.Directed.cs:line 195
   at Robust.Shared.GameObjects.EntitySystem.RaiseLocalEvent[TEvent](EntityUid uid, TEvent& args, Boolean broadcast) in /media/artem/Work/dot-net/space-station-wizards-14/RobustToolbox/Robust.Shared/GameObjects/EntitySystem.cs:line 160
   at Robust.Shared.Physics.Systems.SharedPhysicsSystem.DestroyContact(Contact contact) in /media/artem/Work/dot-net/space-station-wizards-14/RobustToolbox/Robust.Shared/Physics/Systems/SharedPhysicsSystem.Contacts.cs:line 331
   at Robust.Shared.Physics.Systems.SharedPhysicsSystem.CollideContacts() in /media/artem/Work/dot-net/space-station-wizards-14/RobustToolbox/Robust.Shared/Physics/Systems/SharedPhysicsSystem.Contacts.cs:line 502
   at Robust.Shared.Physics.Systems.SharedPhysicsSystem.SimulateWorld(Single deltaTime, Boolean prediction) in /media/artem/Work/dot-net/space-station-wizards-14/RobustToolbox/Robust.Shared/Physics/Systems/SharedPhysicsSystem.cs:line 307
   at Robust.Server.GameObjects.PhysicsSystem.Update(Single frameTime) in /media/artem/Work/dot-net/space-station-wizards-14/RobustToolbox/Robust.Server/GameObjects/EntitySystems/PhysicsSystem.cs:line 34
   at Robust.Shared.GameObjects.EntitySystemManager.TickUpdate(Single frameTime, Boolean noPredictions) in /media/artem/Work/dot-net/space-station-wizards-14/RobustToolbox/Robust.Shared/GameObjects/EntitySystemManager.cs:line 306
   at Robust.Shared.GameObjects.EntityManager.TickUpdate(Single frameTime, Boolean noPredictions, Histogram histogram) in /media/artem/Work/dot-net/space-station-wizards-14/RobustToolbox/Robust.Shared/GameObjects/EntityManager.cs:line 246
   at Robust.Server.GameObjects.ServerEntityManager.TickUpdate(Single frameTime, Boolean noPredictions, Histogram histogram) in /media/artem/Work/dot-net/space-station-wizards-14/RobustToolbox/Robust.Server/GameObjects/ServerEntityManager.cs:line 154
   at Robust.Server.BaseServer.Update(FrameEventArgs frameEventArgs) in /media/artem/Work/dot-net/space-station-wizards-14/RobustToolbox/Robust.Server/BaseServer.cs:line 729
   at Robust.Server.BaseServer.<SetupMainLoop>b__66_1(Object sender, FrameEventArgs args) in /media/artem/Work/dot-net/space-station-wizards-14/RobustToolbox/Robust.Server/BaseServer.cs:line 542
   at Robust.Shared.Timing.GameLoop.Run() in /media/artem/Work/dot-net/space-station-wizards-14/RobustToolbox/Robust.Shared/Timing/GameLoop.cs:line 235
   at Robust.Server.BaseServer.MainLoop() in /media/artem/Work/dot-net/space-station-wizards-14/RobustToolbox/Robust.Server/BaseServer.cs:line 569
   at Robust.Server.Program.ParsedMain(CommandLineArgs args, Boolean contentStart, ServerOptions options) in /media/artem/Work/dot-net/space-station-wizards-14/RobustToolbox/Robust.Server/Program.cs:line 76
   at Robust.Server.Program.Start(String[] args, ServerOptions options, Boolean contentStart) in /media/artem/Work/dot-net/space-station-wizards-14/RobustToolbox/Robust.Server/Program.cs:line 44
   at Robust.Server.ContentStart.Start(String[] args) in /media/artem/Work/dot-net/space-station-wizards-14/RobustToolbox/Robust.Server/ContentStart.cs:line 10
   at Content.Server.Program.Main(String[] args) in /media/artem/Work/dot-net/space-station-wizards-14/Content.Server/Program.cs:line 9
Unhandled exception. Robust.Shared.Utility.DebugAssertException: Exception of type 'Robust.Shared.Utility.DebugAssertException' was thrown.
   at Robust.Shared.Utility.DebugTools.Assert(Boolean condition) in /media/artem/Work/dot-net/space-station-wizards-14/RobustToolbox/Robust.Shared/Utility/DebugTools.cs:line 37
   at Robust.Shared.Physics.Systems.SharedPhysicsSystem.DestroyContacts(PhysicsComponent body) in /media/artem/Work/dot-net/space-station-wizards-14/RobustToolbox/Robust.Shared/Physics/Systems/SharedPhysicsSystem.Components.cs:line 199
   at Robust.Shared.Physics.Systems.SharedPhysicsSystem.OnContactEntityQueueDel(EntityUid obj) in /media/artem/Work/dot-net/space-station-wizards-14/RobustToolbox/Robust.Shared/Physics/Systems/SharedPhysicsSystem.Contacts.cs:line 195
   at Robust.Shared.GameObjects.EntityManager.QueueDeleteEntity(Nullable`1 uid) in /media/artem/Work/dot-net/space-station-wizards-14/RobustToolbox/Robust.Shared/GameObjects/EntityManager.cs:line 627
   at Content.Server.Explosion.EntitySystems.ExplosionSystem.TriggerExplosive(EntityUid uid, ExplosiveComponent explosive, Boolean delete, Nullable`1 totalIntensity, Nullable`1 radius, Nullable`1 user) in /media/artem/Work/dot-net/space-station-wizards-14/Content.Server/Explosion/EntitySystems/ExplosionSystem.cs:line 181
   at Content.Server.Explosion.EntitySystems.TriggerSystem.HandleExplodeTrigger(EntityUid uid, ExplodeOnTriggerComponent component, TriggerEvent args) in /media/artem/Work/dot-net/space-station-wizards-14/Content.Server/Explosion/EntitySystems/TriggerSystem.cs:line 134
   at Robust.Shared.GameObjects.EntityEventBus.<>c__DisplayClass48_0`2.<SubscribeLocalEvent>g__EventHandler|0(EntityUid uid, IComponent comp, TEvent& args) in /media/artem/Work/dot-net/space-station-wizards-14/RobustToolbox/Robust.Shared/GameObjects/EntityEventBus.Directed.cs:line 249
   at Robust.Shared.GameObjects.EntityEventBus.<>c__DisplayClass62_0`1.<EntSubscribe>b__0(EntityUid uid, IComponent comp, Unit& ev) in /media/artem/Work/dot-net/space-station-wizards-14/RobustToolbox/Robust.Shared/GameObjects/EntityEventBus.Directed.cs:line 432
   at Robust.Shared.GameObjects.EntityEventBus.EntDispatch(EntityUid euid, Type eventType, Unit& args) in /media/artem/Work/dot-net/space-station-wizards-14/RobustToolbox/Robust.Shared/GameObjects/EntityEventBus.Directed.cs:line 584
   at Robust.Shared.GameObjects.EntityEventBus.RaiseLocalEventCore(EntityUid uid, Unit& unitRef, Type type, Boolean broadcast) in /media/artem/Work/dot-net/space-station-wizards-14/RobustToolbox/Robust.Shared/GameObjects/EntityEventBus.Directed.cs:line 217
   at Robust.Shared.GameObjects.EntityEventBus.RaiseLocalEvent[TEvent](EntityUid uid, TEvent args, Boolean broadcast) in /media/artem/Work/dot-net/space-station-wizards-14/RobustToolbox/Robust.Shared/GameObjects/EntityEventBus.Directed.cs:line 177
   at Content.Server.Explosion.EntitySystems.TriggerSystem.Trigger(EntityUid trigger, Nullable`1 user) in /media/artem/Work/dot-net/space-station-wizards-14/Content.Server/Explosion/EntitySystems/TriggerSystem.cs:line 234
   at Content.Server.LandMines.LandMineSystem.HandleStepTriggered(EntityUid uid, LandMineComponent component, StepTriggeredEvent& args) in /media/artem/Work/dot-net/space-station-wizards-14/Content.Server/LandMines/LandMineSystem.cs:line 27
   at Robust.Shared.GameObjects.EntityEventBus.<>c__DisplayClass50_0`2.<SubscribeLocalEvent>g__EventHandler|0(EntityUid uid, IComponent comp, TEvent& args) in /media/artem/Work/dot-net/space-station-wizards-14/RobustToolbox/Robust.Shared/GameObjects/EntityEventBus.Directed.cs:line 282
   at Robust.Shared.GameObjects.EntityEventBus.<>c__DisplayClass62_0`1.<EntSubscribe>b__0(EntityUid uid, IComponent comp, Unit& ev) in /media/artem/Work/dot-net/space-station-wizards-14/RobustToolbox/Robust.Shared/GameObjects/EntityEventBus.Directed.cs:line 432
   at Robust.Shared.GameObjects.EntityEventBus.EntDispatch(EntityUid euid, Type eventType, Unit& args) in /media/artem/Work/dot-net/space-station-wizards-14/RobustToolbox/Robust.Shared/GameObjects/EntityEventBus.Directed.cs:line 584
   at Robust.Shared.GameObjects.EntityEventBus.RaiseLocalEventCore(EntityUid uid, Unit& unitRef, Type type, Boolean broadcast) in /media/artem/Work/dot-net/space-station-wizards-14/RobustToolbox/Robust.Shared/GameObjects/EntityEventBus.Directed.cs:line 217
   at Robust.Shared.GameObjects.EntityEventBus.RaiseLocalEvent[TEvent](EntityUid uid, TEvent& args, Boolean broadcast) in /media/artem/Work/dot-net/space-station-wizards-14/RobustToolbox/Robust.Shared/GameObjects/EntityEventBus.Directed.cs:line 195
   at Robust.Shared.GameObjects.EntitySystem.RaiseLocalEvent[TEvent](EntityUid uid, TEvent& args, Boolean broadcast) in /media/artem/Work/dot-net/space-station-wizards-14/RobustToolbox/Robust.Shared/GameObjects/EntitySystem.cs:line 160
   at Content.Shared.StepTrigger.Systems.StepTriggerSystem.OnEndCollide(EntityUid uid, StepTriggerComponent component, EndCollideEvent& args) in /media/artem/Work/dot-net/space-station-wizards-14/Content.Shared/StepTrigger/Systems/StepTriggerSystem.cs:line 171
   at Robust.Shared.GameObjects.EntityEventBus.<>c__DisplayClass50_0`2.<SubscribeLocalEvent>g__EventHandler|0(EntityUid uid, IComponent comp, TEvent& args) in /media/artem/Work/dot-net/space-station-wizards-14/RobustToolbox/Robust.Shared/GameObjects/EntityEventBus.Directed.cs:line 282
   at Robust.Shared.GameObjects.EntityEventBus.<>c__DisplayClass62_0`1.<EntSubscribe>b__0(EntityUid uid, IComponent comp, Unit& ev) in /media/artem/Work/dot-net/space-station-wizards-14/RobustToolbox/Robust.Shared/GameObjects/EntityEventBus.Directed.cs:line 432
   at Robust.Shared.GameObjects.EntityEventBus.EntDispatch(EntityUid euid, Type eventType, Unit& args) in /media/artem/Work/dot-net/space-station-wizards-14/RobustToolbox/Robust.Shared/GameObjects/EntityEventBus.Directed.cs:line 584
   at Robust.Shared.GameObjects.EntityEventBus.RaiseLocalEventCore(EntityUid uid, Unit& unitRef, Type type, Boolean broadcast) in /media/artem/Work/dot-net/space-station-wizards-14/RobustToolbox/Robust.Shared/GameObjects/EntityEventBus.Directed.cs:line 217
   at Robust.Shared.GameObjects.EntityEventBus.RaiseLocalEvent[TEvent](EntityUid uid, TEvent& args, Boolean broadcast) in /media/artem/Work/dot-net/space-station-wizards-14/RobustToolbox/Robust.Shared/GameObjects/EntityEventBus.Directed.cs:line 195
   at Robust.Shared.GameObjects.EntitySystem.RaiseLocalEvent[TEvent](EntityUid uid, TEvent& args, Boolean broadcast) in /media/artem/Work/dot-net/space-station-wizards-14/RobustToolbox/Robust.Shared/GameObjects/EntitySystem.cs:line 160
   at Robust.Shared.Physics.Systems.SharedPhysicsSystem.DestroyContact(Contact contact) in /media/artem/Work/dot-net/space-station-wizards-14/RobustToolbox/Robust.Shared/Physics/Systems/SharedPhysicsSystem.Contacts.cs:line 331
   at Robust.Shared.Physics.Systems.SharedPhysicsSystem.CollideContacts() in /media/artem/Work/dot-net/space-station-wizards-14/RobustToolbox/Robust.Shared/Physics/Systems/SharedPhysicsSystem.Contacts.cs:line 502
   at Robust.Shared.Physics.Systems.SharedPhysicsSystem.SimulateWorld(Single deltaTime, Boolean prediction) in /media/artem/Work/dot-net/space-station-wizards-14/RobustToolbox/Robust.Shared/Physics/Systems/SharedPhysicsSystem.cs:line 307
   at Robust.Server.GameObjects.PhysicsSystem.Update(Single frameTime) in /media/artem/Work/dot-net/space-station-wizards-14/RobustToolbox/Robust.Server/GameObjects/EntitySystems/PhysicsSystem.cs:line 34
   at Robust.Shared.GameObjects.EntitySystemManager.TickUpdate(Single frameTime, Boolean noPredictions) in /media/artem/Work/dot-net/space-station-wizards-14/RobustToolbox/Robust.Shared/GameObjects/EntitySystemManager.cs:line 306
   at Robust.Shared.GameObjects.EntityManager.TickUpdate(Single frameTime, Boolean noPredictions, Histogram histogram) in /media/artem/Work/dot-net/space-station-wizards-14/RobustToolbox/Robust.Shared/GameObjects/EntityManager.cs:line 246
   at Robust.Server.GameObjects.ServerEntityManager.TickUpdate(Single frameTime, Boolean noPredictions, Histogram histogram) in /media/artem/Work/dot-net/space-station-wizards-14/RobustToolbox/Robust.Server/GameObjects/ServerEntityManager.cs:line 154
   at Robust.Server.BaseServer.Update(FrameEventArgs frameEventArgs) in /media/artem/Work/dot-net/space-station-wizards-14/RobustToolbox/Robust.Server/BaseServer.cs:line 729
   at Robust.Server.BaseServer.<SetupMainLoop>b__66_1(Object sender, FrameEventArgs args) in /media/artem/Work/dot-net/space-station-wizards-14/RobustToolbox/Robust.Server/BaseServer.cs:line 542
   at Robust.Shared.Timing.GameLoop.Run() in /media/artem/Work/dot-net/space-station-wizards-14/RobustToolbox/Robust.Shared/Timing/GameLoop.cs:line 235
   at Robust.Server.BaseServer.MainLoop() in /media/artem/Work/dot-net/space-station-wizards-14/RobustToolbox/Robust.Server/BaseServer.cs:line 569
   at Robust.Server.Program.ParsedMain(CommandLineArgs args, Boolean contentStart, ServerOptions options) in /media/artem/Work/dot-net/space-station-wizards-14/RobustToolbox/Robust.Server/Program.cs:line 76
   at Robust.Server.Program.Start(String[] args, ServerOptions options, Boolean contentStart) in /media/artem/Work/dot-net/space-station-wizards-14/RobustToolbox/Robust.Server/Program.cs:line 44
   at Robust.Server.ContentStart.Start(String[] args) in /media/artem/Work/dot-net/space-station-wizards-14/RobustToolbox/Robust.Server/ContentStart.cs:line 10
   at Content.Server.Program.Main(String[] args) in /media/artem/Work/dot-net/space-station-wizards-14/Content.Server/Program.cs:line 9

</details>

**Comment by YuriyKiss** [comment]
> When a regular mine is triggered, after the title, when you step off this mine, an error occurs. There isn't a release mode.

I am aware of that, the problem persists on the engine side, that's why this PR requires https://github.com/space-wizards/RobustToolbox/pull/4883 as stated in the first comment.

By the way, what do you mean by release mode?

**Comment by Arteben** [comment]
> > When a regular mine is triggered, after the title, when you step off this mine, an error occurs. There isn't a release mode.
> 
> I am aware of that, the problem persists on the engine side, that's why this PR requires [space-wizards/RobustToolbox#4883](https://github.com/space-wizards/RobustToolbox/pull/4883) as stated in the first comment.
> 
> By the way, what do you mean by release mode?



> > When a regular mine is triggered, after the title, when you step off this mine, an error occurs. There isn't a release mode.
> 
> I am aware of that, the problem persists on the engine side, that's why this PR requires [space-wizards/RobustToolbox#4883](https://github.com/space-wizards/RobustToolbox/pull/4883) as stated in the first comment.
> 
> By the way, what do you mean by release mode?

Oh, I thought this PR had already been merged
Without release mode it is simple "dotnet run --project Content.Server" for example, with release mode is the command with "-c Release", this mode allow hide some debug errors.

**Review by metalgearsloth — APPROVED** [review]

**Comment by Genkail** [comment]
Another victim of cinema...

**Pull request merged**